src/Integration.UnitTests/MefServices/ActiveSolutionBoundTrackerTests.cs

```diff
@@ -541,5 +559,13 @@ private void VerifyServiceDisconnect(Times expected)
         {
             sonarQubeServiceMock.Verify(disconnectMethod, expected);
         }
+
+        private void VerifyBoundSolutionUiContextIsSet(bool isActive)
+        {
+            var isActiveInt = isActive ? 1 : 0;
+            vsMonitorMock.Verify(x => x.SetCmdUIContext(boundSolutionUiContextCookie, isActiveInt), Times.Once);
+            vsMonitorMock.Verify(x => x.SetCmdUIContext(It.IsAny<uint>(), It.IsAny<int>()), Times.Once);
+            vsMonitorMock.Invocations.Clear();
```

Comment:
This is making a stateful change to the mock which will affect the behaviour of this method when it's called multiple times in the same test.
That's not a problem per se but it's not obvious and it's not how most of the test verification methods work so it deserves to be called out somehow, either by changing the method name or adding a comment (although I can't think of a good name that would do).

Reply:
How's `VerifyAndResetBoundSolutionUiContextMock`? 🤔